### PR TITLE
Fix calc_profit_ratio

### PR DESCRIPTION
**Pull request by stash86**
## Summary

Fix calc_profit_ratio to behave correctly when only either one of amount and open rate is None

Solve the issue: #___

## Quick changelog

- <change log 1>
- <change log 1>

## What's new?

In the description of the function
```
:param amount: Amount to use for the calculation. Falls back to self.amount if not set.
:param open_rate: open_rate to use. Defaults to self.open_rate if not provided.
```
So we can safely assume that when their value isn't given, they will use the current trade's data. But in reality, if we only supply one of them, let's say I do this
```
marked_up_rate = trade.open_rate * 1.3
new_current_profit = trade.calc_profit_ratio(rate=current_rate, open_rate=marked_up_rate)
```
I would expect the bot to fill the amount with trade.amount since I don't supply any to the function, and then calculate the new profif % using the supplied open rate, but it's not. When either one of them is None, the bot will use the original open_trade_value, ignoring the supplied open_rate.

It should only use the original open_trade_value only when both open_rate and amount aren't specified. When only one isn't specified, use the fallback to calculate new open_trade_value

### CODE REVIEW

File: freqtrade/persistence/trade_model.py [hunk 1]
```diff
@@ -1181,15 +1181,19 @@ def calc_profit_ratio(
         """
         Calculates the profit as ratio (including fee).
         :param rate: rate to compare with.
-        :param amount: Amount to use for the calculation. Falls back to trade.amount if not set.
+        :param amount: Amount to use for the calculation. Falls back to self.amount if not set.
```
Comment (xmatthias) [inline freqtrade/persistence/trade_model.py]: I'd leave the comment as `trade.amount` - if you look at this docstring from an IDe completion, `self` won't make sense (or rather, refers to something else) - only `trade.amount` does :)

**Review by xmatthias — COMMENTED** [review]
Does this adress / fix some error or bug you encountered, or is this for "i wanna use this particular function in a strategy" for some simulation? 

---

After a quick look i'm not sure it currently doesn't support empty `amount` (what makes you think this?)
using the current open value is correct in that case (and uses `amount` and `open_rate` implicitly as part of the calculation done when assigning `open_trade_value`

for the `close_trade_value` - it's the same - where `calc_close_trade_value()` will take care of the "fill amount" logic.

**edit** dang i should finish reading the PR comment - not skip to the code early.
We should however add a test for this case - which is clearly an edge-case - one where i'm honestly not sure about what your goal is? (What's the usecase for this)?

**Comment by stash86** [comment]
My usecase is that on a DCA strat, I want to check re-entry triggers based of first entry order's rate. Current implementation is broken. If I only set the rate and open_rate, since the amount is None, it will use open_trade_value that is based of averaged open_rate, not the first order's rate. That's why I was wondering why my extra entries didn't triggered even though manual calc showed otherwise :grin:

Yes, we will need to add test for that.

**Review by xmatthias — APPROVED** [review]
nice :+1:
